### PR TITLE
Send song info

### PR DESCRIPTION
**Pull request by jeinygroove**
This branch combines code from #43 and #26, that's why it has extra commits from #26. I changed Song*.java and some servlets classes just to make my frontend part workable. Now we will send song info object when all analysis finishes successfully and after it we also save it to localStorage, it's important, because when the users refresh the page we won't do the same api requests (for youtube api it's essential, because it has the restriction to the number of search requests).

Rating list displays top songs as well, but it will display all searched songs because at this moment on backend we save all SongInfo's

### CODE REVIEW

**Comment by jeinygroove** [comment]
ATTENTION: Today YouTube API won't work, because the limit of requests per day was exceeded :( You can see how it works and do code review tomorrow

**Review by sashapff — APPROVED** [review]

File: src/main/webapp/src/components/entityAnalysisInfo.js [hunk 1]
```diff
@@ -35,27 +35,35 @@ class EntityAnalysisInfo extends React.Component {
     }
   }
 
+  /** 
+   * Get entity analysis info if it was sended before
```
Comment (jgaillard85) [inline src/main/webapp/src/components/entityAnalysisInfo.js]: It is "sent", not "sended".
Fix in all the places you have used "sended"

File: src/main/java/com/google/alpollo/AnalysisInfoServlet.java [hunk 2]
```diff
@@ -43,6 +43,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     try {
       analysisInfo = gson.fromJson(request.getReader(), AnalysisInfo.class);
       SongDataBase.saveAnalysisInfo(analysisInfo);
+      SongDataBase.saveSongRequest(analysisInfo.getSong());
     } catch (JsonSyntaxException | JsonIOException | IOException e) {
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
```
Comment (amaurymedeiros) [inline src/main/java/com/google/alpollo/AnalysisInfoServlet.java:48]: You're catching the exceptions here to respond with an error, which is the correct thing to do. You can remove the `throws` declaration from the method signature. Same for `doGet`

File: src/main/java/com/google/alpollo/SongDataBase.java [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 import java.util.List;
 import com.google.alpollo.model.AnalysisInfo;
+import com.google.alpollo.model.Song;
+import com.google.alpollo.model.SongCounter;
+import java.util.ArrayList;
```
Comment (amaurymedeiros) [inline src/main/java/com/google/alpollo/SongDataBase.java]: You don't seem to add anything that needs ArrayList. Do you need this import?

File: src/main/java/com/google/alpollo/model/AnalysisInfo.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
 public class AnalysisInfo {
   @Id private Long id;
   /** All the info in this class is attached to a specific parent song. */
-  private final Song song;
+  @Index private Song song;
```
Comment (amaurymedeiros) [inline src/main/java/com/google/alpollo/model/AnalysisInfo.java]: Why did you remove `final`? Is this a requirement for using `@Index`?

File: src/main/webapp/src/components/songInfo.js [hunk 5]
```diff
@@ -94,18 +100,53 @@ class SongInfo extends React.Component {
...
 
+  /** Save state of the component to localStorage. */
+  saveState() {
+    const json = JSON.stringify(this.state);
+    localStorage.setItem("state", json);
+  }
+
+  /** Checks if the state of the component was loaded successfully. */
+  isReady(componentState) {
+      return (componentState && componentState.isLoading === false && componentState.errorMsg === null);
```
Comment (amaurymedeiros) [inline src/main/webapp/src/components/songInfo.js]: wouldn't `!componentState.isLoading` instead of `componentState.isLoading === false` work?

File: src/main/java/com/google/alpollo/model/AnalysisInfo.java (hunk 4 above)
Comment (jeinygroove) [reply src/main/java/com/google/alpollo/model/AnalysisInfo.java]: Yes, objectify don't save `final` fields

File: src/main/java/com/google/alpollo/AnalysisInfoServlet.java (hunk 2 above)
Comment (jeinygroove) [reply src/main/java/com/google/alpollo/AnalysisInfoServlet.java:48]: Redirecting it to @BudiGabe for #43

File: src/main/webapp/src/components/songInfo.js (hunk 5 above)
Comment (jeinygroove) [reply src/main/webapp/src/components/songInfo.js]: hmmm, yes it will work, in this method I was just expecting that state object will definitely have `isLoading` and `errorMsg` fields, that's why I made this check more concrete, but now I see that it's not necessary because of te default state of components, so I'll change it

**Review by jgaillard85 — APPROVED** [review]

File: src/main/java/com/google/alpollo/AnalysisInfoServlet.java (hunk 2 above)
Comment (RazvanBudacaTechquarter) [reply src/main/java/com/google/alpollo/AnalysisInfoServlet.java:48]: Hmm VSCode told me I should throw the IOException too, maybe it's wrong this time.
